### DISCUSSION TITLE
Owner and admin login no longer possible

### DISCUSSION THREAD

**Discussion by DuSisen** [mastodon/mastodon#26852] [Q&A] [answered]
After an upgrade from 4.2.0-beta2 to 4.2.0-beta3, only owner and admin login is no longer possible in the browser. The browser shows: "We're sorry, but something went wrong on our end." All other users (moderators, users, ... and those not logged in) see the website displayed normally (see top-netz.de).

At the time the owner or admin logs in, the log says:

```
Sep 08 12:17:32 elefant bundle[1075]: I, [2023-09-08T12:17:32.424933 #1075]  INFO -- : [d0a0c321-5389-4efb-b205-4f841e771c42] method=POST path=/auth/sign_in format=html controller=Auth::SessionsController action=create status=302 allocations=6012 duration=207.44 view=0.00 db=113.70 location=https://top-netz.de/admin/accounts/110978524791995822
Sep 08 12:17:32 elefant bundle[1075]: I, [2023-09-08T12:17:32.526384 #1075]  INFO -- : [5ca9d569-7ea4-429c-92db-f69a794d3699] method=GET path=/admin/accounts/110978524791995822 format=html controller=Admin::AccountsController action=show status=500 allocations=14759 duration=55.51 view=0.00 db=10.54
Sep 08 12:17:32 elefant bundle[1075]: F, [2023-09-08T12:17:32.527730 #1075] FATAL -- : [5ca9d569-7ea4-429c-92db-f69a794d3699]
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699] ActionView::Template::Error (PG::UndefinedTable: FEHLER:  Relation »software_updates« existiert nicht
Sep 08 12:17:32 elefant bundle[1075]: LINE 1: SELECT "software_updates".* FROM "software_updates"
Sep 08 12:17:32 elefant bundle[1075]:                                          ^
Sep 08 12:17:32 elefant bundle[1075]: ):
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699]     1: - content_for :header_tags do
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699]     2:   = render_initial_state
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699]     3:   = javascript_pack_tag 'public', crossorigin: 'anonymous'
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699]     4:   = javascript_pack_tag 'admin', async: true, crossorigin: 'anonymous'
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699]     5:
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699]
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699] app/models/software_update.rb:33:in `pending_to_a'
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699] app/models/software_update.rb:37:in `urgent_pending?'
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699] app/presenters/initial_state_presenter.rb:13:in `critical_updates_pending'
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699] app/helpers/application_helper.rb:210:in `render_initial_state'
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699] app/views/layouts/admin.html.haml:2
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699] app/views/layouts/admin.html.haml:1
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699] app/controllers/concerns/localized.rb:11:in `set_locale'
Sep 08 12:17:32 elefant bundle[1075]: [5ca9d569-7ea4-429c-92db-f69a794d3699] lib/mastodon/rack_middleware.rb:9:in `call'
```
Did I make a mistake when I did the following to upgrade?
```
bundle config without 'development test'
bundle install -j$(getconf _NPROCESSORS_ONLN)
```
Should I now generate the configuration again? With:
`RAILS_ENV=production bundle exec rake mastodon:setup`

Does the database remain current and in order?

OR much better: can the problem be solved with the admin CLI?

I ask for uncomplicated help.
Thank you in advance.

**Comment by DuSisen** [comment]
I was able to further narrow down the error. There is no database table "software_updates".
I then added "UPDATE_CHECK_URL=" to the .env.production file according to the instructions (https://docs.joinmastodon.org/admin/upgrading/) and restarted Mastodon:

```
systemctl stop mastodon-sidekiq.service mastodon-streaming.service mastodon-web.service
systemctl start mastodon-sidekiq.service mastodon-streaming.service mastodon-web.service
```
The owner and admin were then able to log in to the system again.

Now I see the following message in the owner's dashboard (DE):
"Es gibt ausstehende Datenbankmigrationen. Bitte führe sie aus, um sicherzustellen, dass sich die Anwendung wie erwartet verhält"
Translation:
"There are pending database migrations. Please run them to ensure the application behaves as expected"

Question: Does the following command line resolve this problem easily?

`RAILS_ENV=production bundle exec rails db:migrate`
[Source: https://github.com/mastodon/mastodon/discussions/18412]

An answer is important to me.
Thanks in advance.

**Reply by ClearlyClaire** [reply, marked as answer]
The proper solution is not to disable the update check, but to run database migrations (`RAILS_ENV=production bundle exec rails db:migrate`) and restart the services.

**Reply by DuSisen** [reply]
Thanks for the quick reply. This confirms my suspicion in the last part of my 2nd message / 1st comment. I will now proceed as recommended.

**Reply by DuSisen** [reply]
Thanks again. I did as recommended and also removed `UPDATE_CHECK_URL=` again. My tests and the logs are fine.
